src/script/sign.js

```diff
@@ -0,0 +1,83 @@
+const email_input = document.querySelector("#signin_email");
+const password_input = document.querySelector("#signin_password");
+const error_email = document.querySelector("#error_email");
+const error_password = document.querySelector("#error_password");
+const eyes = document.querySelector(".eye_button");
+
+// 이메일 검사
+function validateEmail() {
+    // 이메일 값이 없는 경우
+    if (email_input.value.trim() === "") {
+        email_input.style.border = "1px solid var(--Linkbrary-red)";
+        error_email.innerText = "이메일을 입력해 주세요.";
+        error_email.style.display = "block";
+    }
+    // 올바른 이메일 주소가 아닐 경우
+    else if (!isValidEmail(email_input.value)) {
+        email_input.style.border = "1px solid var(--Linkbrary-red)";
+        error_email.innerText = "올바른 이메일 주소가 아닙니다.";
+        error_email.style.display = "block";
+    }
+    // 이메일 값이 있는 경우
+    else {
+        email_input.style.border = "1px solid var(--Linkbrary-gray20)";
+        error_email.style.display = "none";
+    }
+}
+
+// 비밀번호 검사
+function validatePassword() {
+    // 비밀번호 값이 없는 경우
+    if (password_input.value.trim() === "") {
+        password_input.style.border = "1px solid var(--Linkbrary-red)";
+        error_password.innerText = "비밀번호를 입력해 주세요.";
+        error_password.style.display = "block";
+    }
+    // 비밀번호 값이 있는 경우
+    else {
+        password_input.style.border = "1px solid var(--Linkbrary-gray20)";
+        error_password.style.display = "none";
+    }
+}
+
+//이메일 유효성 검사
+function isValidEmail(email) {
+    var emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+}
+
+// 로그인
+function login() {
+    if (email_input.value === "test@codeit.com" && password_input.value === "codeit101") {
+        window.location.href = "/folder";
+    } else {
+        if (error_email.style.display === "none") {
+            email_input.style.border = "1px solid var(--Linkbrary-red)";
+            error_email.innerText = "이메일을 확인해 주세요.";
+            error_email.style.display = "block";
+        }
+        if (error_password.style.display === "none") {
+            password_input.style.border = "1px solid var(--Linkbrary-red)";
+            error_password.innerText = "비밀번호를 확인해 주세요.";
+            error_password.style.display = "block";
+        }
+    }
+}
+
+// Enter키로 로그인
+document.addEventListener("keydown", function (event) {
+    if (event.key === "Enter") {
+        login(event);
+    }
+});
+
+// 비밀번호 토글
+function toggleEye() {
+    if (password_input.type === "password") {
+        password_input.type = "text";
+        eyes.style.backgroundImage = "url('../src/images/eye-on.svg')";
+    } else {
+        password_input.type = "password";
+        eyes.style.backgroundImage = "url('../src/images/eye-off.svg')";
+    }
+}
```

Comment:
아래와 같은 식으로 중복 코드를 줄일 수 있겠네요.
다만 이미지를 상대경로로 해두면 이미지 파일이나 js 파일의 위치가 변경될때 에러가 발생할 가능성이 커집니다.

```suggestion
function toggleEye() {
    const inputType = password_input.type === "password" ? "text" : "password";
    const imgUrl = password_input.type === "password" ?  
                "url('../src/images/eye-on.svg')" : "url('../src/images/eye-off.svg')";
                
    password_input.type = inputType
    eyes.style.backgroundImage = imgUrl
}
```

Reply:
그래서 가능하면 이미지 컨트롤은 css를 통해 하시는걸 추천드려요.

```
.eye_button {
    /** 버튼 관련 스타일링**/
    background-image: url('.../eye-open.svg');
}
input[type='password'] + .eye_button {
    ...
    background-image: url('...') // input type 비밀번호일 시 인접 eyebutton 이미지를 eyeClosed.svg로
}

```